tools/engine_tool/lib/src/build_plan.dart

```diff
@@ -361,3 +374,41 @@ enum BuildStrategy {
   const BuildStrategy(this._help);
   final String _help;
 }
+
+typedef _ConfigsByName = Iterable<MapEntry<String, BuilderConfig>>;
+
+/// Computes a list of build configs that can can execute on [environment].
+_ConfigsByName _runnableBuildConfigs(
+  Platform platform, {
+  required Map<String, BuilderConfig> configsByName,
+}) {
+  return configsByName.entries.where((entry) {
+    return entry.value.canRunOn(platform);
+  });
+}
+
+/// Creates a final listing of builds from a mapping of builds.
```

Comment:
FWIW it isn't clear what this does from here.  I'd say `Extracts [Build]s from [runnableConfigs] that can execute on [platform].

Reply:
Done!